R/rdeephaven/R/exports.R

```diff
@@ -7,4 +7,101 @@
+#' ```r
+#' df1 <- data.frame(x=1:10, y=11:20)
+#' th1 <- client$import_table(df1)
+#'
+#' th2 <- th1$
+#'   update("z = x + y")$
+#'   where("z % 4 == 0")
+#' ```
+#' TableHandles also support common functional methods for converting server-side Deephaven tables to R objects stored in
+#' local memory such as `as.data.frame()`, `as_tibble()`, and `as_arrow_table()`. Here's an example of converting the
+#' table created above to an R data frame and verifying that other functional methods work as expected:
+#' ```r
+#' df2 <- as.data.frame(th2)
+#'
+#' print(nrow(th2) == nrow(df2))
+#' print(ncol(th2) == ncol(df2))
+#' print(dim(th2) == dim(df2))
+#' print(all(as.data.frame(head(th2, 2)) == head(df2, 2)))
+#' print(all(as.data.frame(tail(th2, 2)) == tail(df2, 2)))
+#' ```
+#' For more information on these classes and all of their methods, see the reference documentation for [`Client`][Client]
+#' and [`TableHandle`][TableHandle] by clicking on their class names, or by running `?Client` or `?TableHandle`.
+#'
+#' @section
+#' Real-time data analysis:
+#' Since TableHandles are references to tables living on the Deephaven server, they may refer to streaming tables, or
+#' tables that are receiving new data periodically (typically once per second). Here's a simple example of creating a
+#' table that adds a new row every second:
+#' ```r
+#' th3 <- client$time_table("PT1s")$
+#'   update(c("X = ii", "Y = sin(X)"))
+#' ```
+#' R objects like data frames or Dplyr tibbles do not have this streaming property - they are always static objects
+#' stored in memory. However, a TableHandle referring to a streaming table may be converted to a data frame or tibble at
+#' any time, and the resulting object will be a snapshot of the table at the time of conversion. This means that you can
+#' use the Deephaven R Client to perform real-time data analysis on streaming data! Here, we make a simple plot of the
+#' ticking table, and call it three times to demonstrate the dynamic nature of the table:
+#' ```r
+#' plot(as.data.frame(th3)$X, as.data.frame(th3)$Y, type="l")
+#' Sys.sleep(5)
+#' plot(as.data.frame(th3)$X, as.data.frame(th3)$Y, type="l")
+#' Sys.sleep(5)
+#' plot(as.data.frame(th3)$X, as.data.frame(th3)$Y, type="l")
+#' ```
+#' There are performance and memory considerations when pulling data from the server, so it is best to use the provided
+#' TableHandle methods to perform as much of your analysis as possible on the server, and to only pull the data when
+#' something _must_ be done in R, like plotting or writing to a local file.
+#'
+#' @section
+#' Powerful table operations:
+#' Much of the power of Deephaven's suite of table operations is achieved through the use of the [`update_by()`][UpdateBy]
+#' and [`agg_by()`][AggBy] methods. These table methods are important enough to warrant their own documentation pages, accessible
+#' by clicking on their names, or by running `?UpdateBy` or `?AggBy`. These methods come with their own suites of functions,
```

Comment:
It seems weird that `UpdateBy` and `AggBy` are links here as well as `?` operator targets.

If this is as problem, I would search the rest of the docs for similar problems.